site/content/docs/private-locations/checkly-agent-guide.md

```diff
@@ -0,0 +1,162 @@
+
+![private location added](/docs/images/private-locations/private_location_added_command.png)
+
+2) Paste or type the `docker run` command into your container host. Paste your API key from step 6 between the quotation marks for the `API_KEY` environment variable. Optionally, replace `docker` with the command for your container engine of choice (podman, etc.). For example:
+
+`docker run -e API_KEY="pl_...." -d ghcr.io/checkly/agent:latest`
+
+Note that at this point you can add additional environment variables and startup parameters to the run command. You can also set memory and CPU allocations for the container and other settings (e.g. networking) as needed for your container environment (not shown).
+
+3) Optional: Set the `JOB_CONCURRENCY` to the appropriate value (1-10) based on your scaling calculations:
+
+`-e JOB_CONCURRENCY=10`
+
+4) Optional: You can configure an HTTPS or HTTP proxy if one is required for your environment. Add an additional environment variable after the `API_KEY` like this:
+
+`-e HTTPS_PROXY="https://user:password@127.0.0.1:8080"`
+
+5) Run the complete `docker run` command to start the agent.
+
+6) Once the agent container is downloaded and starts up, you can see it in a running state using the appropriate command from your container engine (typically `docker ps`).
+
+```
+CONTAINER ID  IMAGE                         COMMAND        CREATED        STATUS            PORTS   NAMES
+5c8753a42d05  [ghcr.io/checkly/agent:latest](http://ghcr.io/checkly/agent:latest)  node index.js  2 minutes ago  Up 2 minutes ago          confident_leakey
+```
+
+7) You can also check the logs of the new container to ensure it's up and running (typically `docker logs <container_name>`).
+
+```
+[checkly-agent] Starting Consumer c7495186-6f1e-4526-b173-14ee9ad21775
+[checkly-agent] No jobs. Waiting..
+[checkly-agent] No jobs. Waiting..
+[checkly-agent] No jobs. Waiting..
+[checkly-agent] No jobs. Waiting..
+[checkly-agent] No jobs. Waiting..
+```
+
+8) Start additional agents as required. A single agent is fine for testing but multiple agent containers are recommended for redundancy. These agent containers should be on different hosts or nodes.
+
+9) Refresh the private locations page in the Checkly app and you will see a count of the number of running agents.
+
+![agent running](/docs/images/private-locations/agent_running.png)
+
+## Updating the agent container
+
+Since the agents are stateless, they are updated by replacing them or updating the image in place. An in-place upgrade is easiest as it keeps the previously defined environment variables. You can use this command to do an in-place upgrade of an agent container. Ensure you have sufficient agent capacity as the agent container will have a short outage as it is upgraded. As agent shutdowns are graceful, no running checks will be lost:
+
+`docker run --rm -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once <container-name-to-be-updated>`
```

Comment:
It might make sense to clarify that this is a third-party tool and link to https://containrrr.dev/watchtower/

I also think it could be more explicit that this is optional. Some users might prefer to just update the agent version in their IaC config file.

Reply:
Fixed.